File: haskell/difference-of-squares/src/Squares.hs
module Squares
  ( difference
  , squareOfSums
  , sumOfSquares
  ) where

difference :: Integral a => a -> a
difference n = squareOfSums n - sumOfSquares n

squareOfSums :: Integral a => a -> a
squareOfSums n = sum [1 .. n] ^ 2

sumOfSquares :: Integral a => a -> a
sumOfSquares n = sum $ map (^ 2) [1 .. n]

File: exercism/kotlin/difference-of-squares/src/main/kotlin/Squares.kt
class Squares(val num: Int)  {

    fun squareOfSum(): Int {
        val sum = (1..num).sum()
        return square(sum)
    }

    fun sumOfSquares(): Int {
        return (1..num).map(::square).sum()
    }

    fun difference(): Int {
        return squareOfSum() - sumOfSquares()
    }
}

private fun square(n: Int): Int {
    return n * n
}
